### ISSUE TITLE
amended .rda files disappear from commit in RStudio Git History

### ISSUE THREAD

**Issue by lindbrook** [rstudio/rstudio#13126]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2023.03.1+446
    OS Version      : macOS 13.3.1(a)
    R Version       : 4.3.0 (2023-04-21)

### Steps to reproduce the problem

1) amend an .rda file in your package.
2) use usethis::use_data(overwrite = TRUE) to update file
3) add files to commit
4) push to GitHub
5) go to Git:History panel in RStudio (RStudio Review Changes window)

-> the amended files are no longer listed there (as if you never committed them)

### Describe the problem in detail

I noticed that, with at least RStudio (2023.03.1+446) on macOS, amended .rda files "disappear" from the commit in Git:History (RStudio Review Changes window) after you push them. In the example below, latlong.ortho.pump.vestry.rda and latlong.ortho.pump.rda, are not visible:

![image](https://github.com/rstudio/rstudio/assets/6076793/c307c0e8-8585-4f4f-9dc5-74488ab2bfb9)

At first, I thought I simply forgot to add the changed files to the commit (this is the second time I've seen this behavior) but you can seem the changes on GitHub in the Commit and Files changed tabs as BIN files: https://github.com/lindbrook/cholera/pull/223

And, after merging and pulling they are available on my computer. In short, everything seems to be working but the changes are just not visible in RStudio's Git History.

### Describe the behavior you expected

Amended .rda files should appear in Git:History panel in RStudio. 

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by lindbrook** [comment]
FWIW, same result when committing new .rda binary files.

**Comment by kevinushey** [comment]
Presumedly the issue here arises with any kind of binary file. Here's what git is printing:

```
/usr/bin/git -c core.quotepath=false diff 1e39d78a91b93898a2c463f367bff049c3928207^ 1e39d78a91b93898a2c463f367bff049c3928207
diff --git data.rds data.rds
index 003bef0..d8157b1 100644
Binary files data.rds and data.rds differ
```

Compared to a regular text file:

```
$ /usr/bin/git -c core.quotepath=false diff 1d934e885d5c836e10bd63266aa848229fc9ebec^ 1d934e885d5c836e10bd63266aa848229fc9ebec
diff --git hello.R hello.R
new file mode 100644
index 0000000..d81cc07
--- /dev/null
+++ hello.R
@@ -0,0 +1 @@
+42
```


We'll probably have to update the parser in https://github.com/rstudio/rstudio/blob/main/src/gwt/src/org/rstudio/studio/client/workbench/views/vcs/common/diff/UnifiedParser.java to handle this.

**Comment by lindbrook** [comment]
Curious. Once fixed, will the "missing" files of past commits then appear in the RStudio Git:History panel?

**Comment by kevinushey** [comment]
Yes, they will. For context, the issue here was only in the UI / presentation of the Git history in RStudio; the repository state itself is not affected.

**Comment by jonvanausdeln** [comment]
Verified on 2023.09.0-daily+290
Windows 11, MacOS 12.6

Binary files show up as expected.

@lindbrook , if you'd like to try out the fix it is available in the latest daily build [2023.09.0-daily+290](https://dailies.rstudio.com/version/2023.09.0-daily+290). Please note that daily builds are intended for testing purposes, and are not recommended for general use.
